lib/Doctrine/ODM/PHPCR/Translation/TranslationStrategy/AttributeTranslationStrategy.php

```diff
@@ -173,9 +173,9 @@ public function removeAllTranslations($document, NodeInterface $node, ClassMetad
     public function getLocalesFor($document, NodeInterface $node, ClassMetadata $metadata)
     {
         $locales = array();
-        foreach ($node->getProperties("*{$this->prefix}*") as $prop) {
+        foreach ($node->getProperties($this->prefix . ':*') as $prop) {
             $matches = null;
-            if (preg_match('/' . $this->prefix . ':(..)-[^-]*/', $prop->getName(), $matches)) {
+            if (preg_match('/' . $this->prefix . ':([a-zA-Z1-9_]+)-[^-]*/', $prop->getName(), $matches)) {
```

Comment:
is this correct? the more i look at this the more i get confused. what is the point of the `[^-]*` part? matching anything after the minus that is not a minus, but its optional and not even used afterwards... i think the original code was wrong here.

the new regex however will only match letters and numbers, and only if they are followed by _. the _ is optional. actually we should just check for something that comes after the prefix. and we have a naming issue in there - the thing called prefix can only be a namespace alias, not any prefix, because the `:` is hardcoded all over the place.